src/GraphQL/Language/AST/CommentNode.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+
+namespace GraphQL.Language.AST
+{
+    public class CommentNode : AbstractNode
+    {
+        protected string _comment;
+
+        public CommentNode(string comment)
+        {
+            _comment = comment;
+        }
+
+        public new string Comment => _comment;
```

Comment:
Is it required? With this property one may get the impression that you can set a comment on the comment. I guess that here you wanted to express this property as `Value` but I doubt that anyone needs this explicit property, since all work will go through `AbstractNode.Comment`.

Reply:
I did rename this property as Value but we need it because we need a way to get the actual text value of a comment through AbstractNode.Comment :
`public string Comment => CommentNode?.Value;`
If I remove it entirely, some tests are failing.
But maybe I get you wrong.

Reply:
OK